custom-widget-builder/src/HtmlTemplatesGenerator.ts

```diff
@@ -0,0 +1,232 @@
+      for (let valIndex = 0; valIndex < combination.length; valIndex++) {
+        // One boolean, e.g. 'true'
+        let booleanVal = combination[valIndex];
+        if (valIndex !== 0) {
+          ifElem.push(' && ');
+        }
+        ifElem.push(`${booleanVal ? '' : '!'}properties.${booleanProps[valIndex].name}`);
+      }
+      ifElem.push('"\n\t');
+      // Iteration to get the required boolean properties
+      for (let valIndex = 0; valIndex < combinations[combIndex].length; valIndex++) {
+        let combination = combinations[combIndex];
+        let booleanVal = combination[valIndex];
+        if (booleanVal) {
+          ifElem.push(`${booleanProps[valIndex].name}\n\t`);
+        }
+      }
+      ifArray.push(ifElem.join(''))
+    }
+    return ifArray;
+  }
+
+  private generateSpecificProperties(type: FwkType): string {
+    let specificProps = "";
+    for (let prop of this.propertiesInfo.properties) {
+      if (!HtmlTemplatesGenerator.isSpecific(prop)) {
+        continue;
+      }
+      if (type === FwkType.AngularJS) {
+        specificProps += `ng-${prop.name.toLowerCase()}="properties.${prop.name}"\n\t`
+      } else if (type === FwkType.Angular) {
+        specificProps += `[${prop.name.toLowerCase()}]="properties.${prop.name}"\n\t`
+      }
+    }
+    return specificProps;
+  }
+
+  private generateProperties(type: FwkType): string {
+    let props = "";
+    for (let prop of this.propertiesInfo.properties) {
+      if (HtmlTemplatesGenerator.isToIgnore(prop) ||
+        HtmlTemplatesGenerator.isSpecific(prop) ||
+        HtmlTemplatesGenerator.isBoolean(prop)) {
+        continue;
+      }
+      if (type === FwkType.Angular && prop.bond === Bond.Variable) {
+        props += `[(ngModel)]="properties.${prop.name}"\n\t`;
+      } else {
+        props += `${prop.name}="{{properties.${prop.name}}}"\n\t`;
+      }
+    }
+    return props;
+  }
+
+  private generateHtmlTagEnd(): string {
+    return `></${this.propertiesInfo.name}>\n`;
+  }
+
+  private static isSpecific(prop: Property): boolean {
+    return HtmlTemplatesGenerator.specificProperties.includes(prop.name);
+  }
+
+  private static isToIgnore(prop: Property): boolean {
+    return HtmlTemplatesGenerator.toIgnoreProperties.includes(prop.name);
+  }
+
+  private static isBoolean(prop: Property) {
+    return prop.type === "boolean";
+  }
+
+  private static booleanComb(nb: number): Array<Array<boolean>> {
+    let globalArray: Array<Array<boolean>> = [];
+    if (nb === 0) {
+      return globalArray;
+    }
+    for (let ii = 0; ii < (1 << nb); ii++) {
+      let boolArr = [];
+      for (let jj = 0; jj < nb; jj++) {
+        boolArr.push(Boolean(ii & (1 << jj)));
+      }
+      globalArray.push(boolArr);
+    }
+    return globalArray;
+  }
+
+  private writeFile(template: string, type: FwkType) {
+    let ext;
+    if (type === FwkType.AngularJS) {
+      ext = "tpl.html";
+    } else if (type === FwkType.Angular) {
+      ext = "tpl.runtime.html";
```

Comment:
Use static variable define to avoid typo and future uptade

Reply:
yes obviously !